src/rdf4cpp/rdf/datatypes/registry/DatatypeIRI.hpp

```diff
@@ -0,0 +1,126 @@
+#ifndef RDF4CPP_REGISTRY_DATATYPEIRI_HPP
+#define RDF4CPP_REGISTRY_DATATYPEIRI_HPP
+
+#include <variant>
+#include <cstdint>
+#include <functional>
+
+#include <rdf4cpp/rdf/storage/node/identifier/LiteralType.hpp>
+
+namespace rdf4cpp::rdf::datatypes::registry {
+
+template<typename F, typename O>
+struct DatatypeIRIVisitor {
+    F map_fixed;
+    O map_other;
+
+    auto operator()(storage::node::identifier::LiteralType const fixed) -> std::invoke_result_t<F, storage::node::identifier::LiteralType> {
+        return std::invoke(this->map_fixed, fixed);
+    }
+
+    template<typename S>
+    auto operator()(S &&other) -> std::invoke_result_t<O, decltype(std::forward<S>(other))> {
+        return std::invoke(this->map_other, std::forward<S>(other));
+    }
+};
+
+template<typename F, typename O>
+DatatypeIRIVisitor(F, O) -> DatatypeIRIVisitor<F, O>;
+
+struct DatatypeIRIView {
```

Comment:
Probably need a better name for DatatypeIRIView and DatatypeIRI. Maybe you have ideas? (and classdoc missing)

Reply:
I tend to say that just DatatypeIRI would be fine. 

I did not get why in there is those two [DatatypeIRIView](https://github.com/rdf4cpp/rdf4cpp/blob/feature/fixed-datatype-ids/src/rdf4cpp/rdf/datatypes/registry/DatatypeIRI.hpp#L30) and [DatatypeIRI](https://github.com/rdf4cpp/rdf4cpp/blob/feature/fixed-datatype-ids/src/rdf4cpp/rdf/datatypes/registry/DatatypeIRI.hpp#L68), though.

Reply:
There is one owned version that can contain `std::string` (DatatypeIRI) and one borrowing version that can contain `std::string_view` (DatatypeIRIView)

Reply:
After the review, I would vote for `DatatypeIdentifier` / `DatatypeIdentifierView`